daskernetes/core.py

```diff
@@ -55,69 +82,96 @@ class KubeCluster(object):
     """
     def __init__(
             self,
+            worker_spec=None,
             name=None,
             namespace=None,
-            worker_image='daskdev/dask:latest',
-            worker_labels=None,
+            labels=None,
             n_workers=0,
-            threads_per_worker=1,
             host='0.0.0.0',
-            port=8786,
+            port=0,
             env={},
             **kwargs,
     ):
-        self.cluster = LocalCluster(ip=host or socket.gethostname(),
-                                    scheduler_port=port,
-                                    n_workers=0, **kwargs)
-
-        try:
-            config.load_incluster_config()
-        except config.ConfigException:
-            config.load_kube_config()
-
-        self.api = client.CoreV1Api()
-
         if namespace is None:
             namespace = _namespace_default()
+        self.namespace = namespace
 
         if name is None:
             name = 'dask-%s-%s' % (getpass.getuser(), str(uuid.uuid4())[:10])
 
-        self.namespace = namespace
-        self.name = name
-        self.worker_image = worker_image
-        self.worker_labels = (worker_labels or {}).copy()
-        self.threads_per_worker = threads_per_worker
-        self.env = dict(env)
+        labels = dict(labels or {})
+        env = dict(env)
 
         # Default labels that can't be overwritten
-        self.worker_labels['org.pydata.dask/cluster-name'] = name
-        self.worker_labels['app'] = 'dask'
-        self.worker_labels['component'] = 'dask-worker'
+        labels['org.pydata.dask/cluster-name'] = name
+        labels['app'] = 'dask'
+        labels['component'] = 'dask-worker'
+        self.labels = labels
+
+        if worker_spec is None:
+            try:
+                worker_spec = config['worker']['spec']
+            except KeyError:
+                worker_spec = make_worker_spec()
+
+        worker_spec = deserialize_pod_spec(worker_spec)
+        self.worker_spec = worker_spec
+
+        self.worker_meta = client.V1ObjectMeta(
+                generate_name=name + '-',
+                labels=labels
+        )
```

Comment:
Is this a sensible approach?  To get a pod spec from the user and create metadata ourselves?